### ISSUE TITLE
Permissions issue. Only "Others" permission taken into account for Nginx + PHP-FPM

### ISSUE THREAD

**Issue by rkkoszewski** [hestiacp/hestiacp#635]
### In a few words, please describe the issue that you're experiencing:

All .PHP files need the "others" permission to be set to "read" or the script itself has no permission to access other .PHP files from the same domain owner (Even when the user has permission to access that file).

In other words:
Domain user: user1
Web root:
a.php user1:user1 777
b.php user1:user1 770

When a.php tries to import b.php it throws an exception saying it has no permission to access that file.

### What steps did you take when the issue occured? 
Create a new user in Hestia CP.

Create a file called index.php in the public_html folder of that user with permissions of 777. In the index.php file use import to import test.php. Create a file called test.php in the same directory that echos some string, and that file uses 770 as permission. Both files have the owner and group set to the domain user (not root).

Now open index.php from your web browser and you get an exception because index.php is unable to read test.php.
Open test.php and you get an 404 (expected)

### Expected behavior:
Index.php should be able to access test.php as php-fpm is running as the domain user. Test.php should still return 404 when accessed directly (All good there)

### Operating system distribution and release:
Debian 9.11

### Which version of Hestia Control Panel is currently installed?

Latest

### Which branch are you using?
Debian install script

### When did this issue occur? After a clean installation, or after an upgrade?
Please enter your response here (e.g the feature stopped working after upgrading to the latest release)

It's a fresh install.

### Which of the following web server configurations are you using?
- Nginx + PHP-FPM + Multi-PHP

### Which of the following server roles are installed on your server?
- DNS
- Mail (Dovecot + Exim)
- MySQL (MariaDB)

### Additional notes:
To give an example where this issue is present for me. I'm running Nextcloud and I have to give to all source files at least the "read" permission for "others", for every single file that Nextcloud has (Even the configuration files). Otherwise I get an exception due to permissions. When there is a Nexcloud update the permissions get reset 660 for all files, breaking the website with every update.

But worst of all is that every user on the system, when they FTP or SSH into the server, they can read absolutely all files from other users (when they know the path, which is not difficult to know). So every user can read the configuration files of everyone on the server like that. I don't think this is supposed to be like this, unless I'm wrong here.

**Comment by Lupul** [comment]
There is something wrong here, by default php-fpm pools are running as a Hestia user so there is no need change any permissions. 
I've just tested your scenario with 777&770 as you sugested and no problem, even chmod 0600 didn't throw any errors. (debian9+nginx+multiphp)

As a side note do not set these types of perms (777: word writable and executable)
- 640 are enough for php scripts
- 644 for static files accessed by the webserver
- 66* to grant write perms to the group.
- 7** only for folders and executable scripts or binary files

**Comment by Lupul** [comment]
On your second point, two Hestia users cannot read each others files no mater what perms they set to their files. (access is limited trough posix ACL)

**Comment by ghost** [comment]
Can you please clarify which version are you running specifically, 1.0.6 from the release branch or a 1.1.0 build from the master/development branch?

**Comment by rkkoszewski** [comment]
@kristankenney 
I'm running Hestia CP 1.0.6 on a Debian 9.11 (64bit) in an LXC container. Kernel 5.0.21-5-pve #1 SMP PVE 5.0.21-10 (Wed, 13 Nov 2019 08:27:10 +0100) x86_64 GNU/Linux.

As the setup I installed a fresh Debian 9, updated all packages and installed Hestia CP. No exceptions during install and everything seems to be working fine outside of the permission issue. 

When I request a page from the user "a", I can see a php-fpm process is spun up as user "a", and in the domains pool config file I can see that user and group is set to user "a", the listen.owner is "a" and listen.group is  "www-data".

So I'm completely puzzled. If the PHP process is running as user "a" and it's trying to run PHP files with owner a:a with permission 770, why does it fail with permission denied untill I set the "other" permission?

**Comment by rkkoszewski** [comment]
Ok, I just checked with my old Vesta CP setup and I see that the "other" permissions seem to be the same like now. Seems like the default file permissions during FTP upload have the "other" permission set to 0 by default which lead to my confusion. 

With most of the hosting providers that I used had a setup where when setting the "other" permission would allow the script to be executed by the world, while all the scripts (and files) that did not have the "other" permission could still be included as long as the domain user had permissions. 

Closing the issue as it is no other than what was happening in VestaCP. I might reopen if I find out some more info about this settings.
